tests/unit-tests/test_dataloading.py

```diff
@@ -1306,7 +1313,32 @@ def test_lp_dataloader_len(batch_size):
                                                device='cuda:0', train_task=True)
     assert len(dataloader) == len(list(dataloader))
 
+@pytest.mark.parametrize("num_pos", [2, 10])
+@pytest.mark.parametrize("num_neg", [5, 20])
+def test_inbatch_joint_neg_sampler(num_pos, num_neg):
+    src = th.arange(num_pos)
+    dst = th.arange(num_pos)
+    g = dgl.heterograph({
+        ("n0", "r0", "n1"): (src, dst),
+    })
+    sampler = InbatchJointUniform(num_neg)
+    src, dst = sampler._generate(g, th.arange(num_pos), ("n0", "r0", "n1"))
+    assert len(src) == num_pos * (num_pos - 1) + num_pos * num_neg
+    assert len(dst) == num_pos * (num_pos - 1) + num_pos * num_neg
+    in_batch_src = src[-num_pos * (num_pos - 1):]
```

Comment:
Why do we need to use this method to get in-batch source and destinations? 

I thought the outputs of  sampler._generate() is the in-batch source and destination.

Reply:
I see. It is concat in the _generate() method.